### DISCUSSION TITLE
Is there a way to invalidate a data fetching cache key from a page that didn't use it?

### DISCUSSION THREAD

**Discussion by whichken** [nuxt/nuxt#17515] [Questions]
So the whole data fetching caching stuff is cool and seems useful to increase the snappiness of apps, but it seems like I keep shooting myself in the foot when I try to make use of it.

For example, this flow works great:
- I use `useAsyncData` with a custom key to fetch some data and use it to display an element on the page.
- I make a POST or PUT call to change the data.
- I use the `refresh` method or `refreshNuxtData` with that custom key.

However, this flow doesn't work:
- I use `useAsyncData` with a custom key to fetch some data and use it to display an element on the page.
- I route the user to a separate edit page and make a POST or PUT call from there.
- I call the `refreshNuxtData` with that custom key to invalidate the cache.
- I route the user back to the original page.

As far as I can tell, the `refreshNuxtData` method doesn't invalidate the cache if the request you're trying to invalidate didn't occur on that page. This means the user sees the original outdated data when routed back to the original page.

Is there a way to invalidate this cache by key from anywhere in the app (not just from the page that made the request)? I am currently just migrating all of the problem data calls over to $fetch so that they don't get cached. This solves my problem, but it means I don't get the benefit of caching in places where it could help us.

**Comment by lacorde** [comment]
I agree with you, and I would go even further by saying that, in your first example, **the cache isn't invalidated either.**
If you leave the page and come back to it, the old data is still there.

So is **refreshNuxtData** just a display trick ? Should we open an issue about this ?

The only way I found to get around caching issues is setting `initialCache` to `false` on my `useAsyncData` calls.

**Comment by cawa-93** [comment]
After studying the [source code](https://github.com/nuxt/framework/blob/ecc8c0adb944791dd8a8236274faf9584dddf929/packages/nuxt/src/app/composables/asyncData.ts#L100-L102), it looks like you can clear cache by key as follows:
```ts
  const nuxt = useNuxtApp();
  const key = 'list';
  if (nuxt.payload?.data?.[key] !== undefined) {
    nuxt.payload.data[key] = undefined;
    nuxt.payload._errors[key] = null;
  }
```

https://stackblitz.com/edit/github-rqe159?file=pages/add.vue

**Reply by cawa-93** [reply]
I'll try to add a new compostable function for this later

**Reply by cawa-93** [reply]
https://github.com/nuxt/framework/pull/5227
